src/winetricks

```diff
@@ -614,6 +614,68 @@ w_dotnet_verify()
             dn_status=$?
 }
 
+# Checks if the user can run the self-update/rollback commands
+winetricks_check_update_availability()
+{
+    # Prevents the development file overwrite:
+    if test -d "../.git"
+    then
+        w_warn "You're running in a dev environment. Please make a copy of the file before running this command."
+        exit;
+    fi
+
+    # Checks read/write permissions on update directories
+    if ! (test -r $0 && test -w $0 && test -w ${0%/*} && test -x ${0%/*})
+    then
+        w_warn "You don't have the proper permissions to run this command. Try again with sudo or as root."
+        exit;
+    fi
+}
+
+winetricks_selfupdate()
+{
+    winetricks_check_update_availability
+
+    _W_filename="${0##*/}"
+    _W_rollback_file="${0}.bak"
+    _W_update_file="${0}.update"
+
+    _W_tmpdir=${TMPDIR:-/tmp}
+    _W_tmpdir="`mktemp -d "$_W_tmpdir/$_W_filename.XXXXXXXX"`"
+
+    w_download_to $_W_tmpdir https://raw.githubusercontent.com/Winetricks/winetricks/master/src/winetricks
+    mv $_W_tmpdir/$_W_filename $_W_update_file.gz
+    gunzip $_W_update_file.gz
```

Comment:
Please add w_try around shell commands (here and below).

Looks fine other than that.